R/copernicus_article.R

```diff
@@ -0,0 +1,129 @@
+#'
+#' An number of required and optional manuscript sections, e.g. \code{acknowledgements}, \code{competinginterests}, or \code{authorcontribution}, must be declared using the respective properties of the R Markdown header - see skeleton file.
+#'
+#' \strong{Version:} Based on copernicus_package.zip in the version 5.0, 21 March 2018, using \code{copernicus.cls} in version 8.67, 30 January 2018
+#'
+#' \strong{Important note:} The online guidelines by Copernicus are the official resource.
+#' Copernicus is not responsible for the community contributions made to support the template in this package.
+#' Copenicus converts all typeset TeX files into XML, the expressions and markups have to be highly standardized.
+#' Therefore, please keep the following in mind:
+#'
+#' \itemize{
+#'   \item Please provide only one figure file for figures with several panels, and please do not use \code{\\subfloat} or similar commands.
+#'   \item Please use only commands in which words, numbers, etc. are within braces (e.g. \code{\\textrm{TEXT}} instead of \code{{\\rm TEXT}}).
+#'   \item For algorithms, please use the syntax given in template.tex or provide your algorithm as a figure.
+#'   \item Please do not define new commands.
+#'   \item The most commonly used packages (\code{\\usepackage{}}) are integrated in the copernicus.cls. Some other packages often used by the community are defined in template.tex. Please do not insert additional ones in your *.tex file.
+#'   \item Spaces in labels (\code{\\label{}}) are not allowed; please make sure that no label name is assigned more than once.
+#'   \item Please do not use \code{\\paragraph{}}; only \code{\\subsubsection{}} is allowed.
+#'   \item It is not possible to add tables in colour.
+#' }
+#'
+#' @note If you use \code{rmarkdown::pdf_document()}, all internal references (i.e. tables and figures) must use \code{\\ref\{\}} whereas with \code{bookdown::pdf_document2()}, you can additionally use \code{\\@@ref()}.
+#'
+#' @references
+#' Manuscript preparation guidelines for authors.
+#' \url{https://publications.copernicus.org/for_authors/manuscript_preparation.html}
+#'
+#' @examples
+#'
+#' \dontrun{
+#' library("rmarkdown")
+#' draft("MyArticle.Rmd", template = "copernicus_article", package = "rticles")
+#' render("MyArticle/MyArticle.Rmd")
+#' }
+#'
+#' @export
+copernicus_article <- function(...,
+                         keep_tex         = TRUE,
+                         citation_package = "natbib",
+                         base_format = rmarkdown::pdf_document,
+                         md_extensions = c(
+                           "-autolink_bare_uris", # disables automatic links, needed for plain email in \correspondence
+                           "-auto_identifiers"    # disables \hypertarget commands
+                           )) {
+  if (inherits(base_format, "character")) {
```

Comment:
How about `is.character()`?

Reply:
Just copied this from `peerj_article`, but happy to change to `is.character()`.